external/concealer/src/autoware.cpp

```diff
@@ -144,97 +97,9 @@ bool Autoware::ready() const
   return task_queue.exhausted();
 }
 
-/* ---- NOTE -------------------------------------------------------------------
- *
- *  Send initial_pose to Autoware.
- *
- * -------------------------------------------------------------------------- */
-void Autoware::initialize(const geometry_msgs::msg::Pose & initial_pose)
+void Autoware::createUpdater()
```

Comment:
Unfortunately, I cannot initialize `updater` in `Autoware::Autoware()` initializer list. `updater` creates wall timer, that periodically calls `update()` method that is purely virtual. The program crashes with `call to the purely virtual method` error. `updater` must be initialized after `update()` method from derived class is assigned to vtable. The only way to do this is to call `createUpdater()` in body of derived class constructor. So `Autoware::updater` cannot be `const` for this reason.

Therefore, the only change I can make is to make `createUpdater()` return the result of `create_wall_timer(...)`, but I am not sure if this is valuable without other changes you suggested. Would you like me to do this?

Reply:
On the other hand, I think I can just make `update()` an empty virtual function instead of purely virtual. I think it would solve the problem as well

Reply:
I think your idea is good.

A nop virtual function is a bit less beautiful, but I think it's safer because the updater becomes constant and doesn't require classes derived from Autoware to call createUpdater.